### PR TITLE
.Net: make planner constructors more similar

### LINKED ISSUE #2847: Planner Constructor normalization

### PR DESCRIPTION
**Pull request by hario90**
### Motivation and Context

Fixes #2847,  #2519

Contains breaking changes.

This addresses inconsistencies in planner construction. The planners sometimes took a prompt override through a delegate in the config and sometimes as a constructor argument. Additionally, the ActionPlanner used to take a loggerFactory as a constructor argument while the other planners did not. 

The goal of these changes is to make working with planners more predictable.

### Description

* Added `GetPromptTemplate` delegate function to the planner base config. All planners will use this to get the prompt override.
* Removed logger factory argument from ActionPlanner constructor. Used the Kernel's logger factory to create a new logger within each planner constructor.
* Renamed private `Config` private instance field to `_config` 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs [hunk 1]
```diff
@@ -24,25 +24,25 @@ public sealed class SequentialPlanner : ISequentialPlanner
...
+        // Set up prompt template
+        string promptTemplate = this._config.GetPromptTemplate?.Invoke() ?? EmbeddedResource.Read("skprompt.txt");
 
         this._functionFlowFunction = kernel.CreateSemanticFunction(
             promptTemplate: promptTemplate,
             skillName: RestrictedSkillName,
             description: "Given a request or command or goal generate a step by step plan to " +
                          "fulfill the request using functions. This ability is also known as decision making and function flow",
-            maxTokens: this.Config.MaxTokens ?? 1024,
+            maxTokens: this._config.MaxTokens ?? 1024,
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs:45]: This stuff is going to be changing with the `AIRequestSettings` work. Just a heads up.

**Review by lemillermicrosoft — APPROVED** [review]

**Comment by shawncal** [comment]
@hario90 We've got some conflicts. Can you please resolve, then ping me for final review and merge?

File: dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ public async Task<Plan> CreatePlanAsync(string goal, CancellationToken cancellat
         return plan;
     }
 
-    private SequentialPlannerConfig Config { get; }
+    private SequentialPlannerConfig _config { get; }
```
Comment (SergeyMenshykh) [inline dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs:97]: nit: The property was renamed according to the field's rules. Its name follows camel case, and it is prefixed with `_`. So, if the intention of this change is to convert the property into a class field, please remove the remaining getter.

File: dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs [hunk 3]
```diff
@@ -682,7 +682,7 @@ private static string ToFullyQualifiedName(FunctionView function)
     /// <summary>
     /// The configuration for the StepwisePlanner
     /// </summary>
-    private StepwisePlannerConfig Config { get; }
+    private StepwisePlannerConfig _config { get; }
```
Comment (SergeyMenshykh) [inline dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs:685]: nit: The property was renamed according to the field's rules. Its name follows camel case, and it is prefixed with _. So, if the intention of this change is to convert the property into a class field, please remove the remaining getter.

Comment (hario90) [reply dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs:685]: I confused this with an instance field - I'll change this back.

File: dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs (hunk 2 above)
Comment (hario90) [reply dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs]: fixed - thanks!

**Review by SergeyMenshykh — APPROVED** [review]

**Comment by banduki** [comment]
Is `GetPromptTemplate()` a delegate to allow override of the create plan prompt? -- or if not, what is it used for?

**Pull request merged**